examples/backpack_example.py

```diff
@@ -0,0 +1,293 @@
+import logging
+import os
+from dataclasses import dataclass
+from typing import Optional
+
+import equinox as eqx
+import jax.random as jrandom
+import jmp
+from jax.sharding import PartitionSpec
+from transformers import GPT2Tokenizer
+
+import haliax as hax
+import haliax.random
+import levanter
+import wandb
+from haliax import Axis
+from haliax.partitioning import ResourceAxis, named_jit, round_axis_for_partitioning
+from levanter import callbacks
+from levanter.config import TrainerConfig
+from levanter.data.sharded import GlobalBatchDataset, LocalBatchDataset
+from levanter.data.text import LMDatasetConfig, TokenSeqDataset
+from levanter.grad_accum import accumulate_gradients_sharded
+from levanter.logging import capture_time, log_time_to_wandb
+from levanter.models.backpack import BackpackConfig, BackpackLMHeadModel
+from levanter.models.loss import next_token_loss
+from levanter.trainer_hooks import StepInfo, TrainerHooks
+from levanter.utils.jax_utils import global_key_array, parameter_count
+from levanter.utils.py_utils import non_caching_cycle
+
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class TrainBackpackConfig:
```

Comment:
other than this class, is it total copypaste? (fine if so, just won't look as carefully)

Reply:
I pasted it into gpt2_example.py. It is mostly copypaste. We definitely merge these two files, to save cost of maintenance in the future. I might do it in a second PR.